src/hdmf/common/sparse.py

```diff
@@ -22,32 +22,37 @@ def __init__(self, **kwargs):
         data = getargs('data', kwargs)
         if isinstance(data, (np.ndarray, h5py.Dataset)):
             if data.ndim == 2:
-                data = sps.csr_matrix(self.data)
-            elif data.ndim == 1:
+                data = sps.csr_matrix(data)
+            elif data.ndim < 2:
                 indptr, indices, shape = getargs('indptr', 'indices', 'shape', kwargs)
                 if any(_ is None for _ in (indptr, indices, shape)):
-                    raise ValueError("must specify indptr, indices, and shape when passing data array")
-                self.__check_ind(indptr, 'indptr')
-                self.__check_ind(indices, 'indices')
+                    raise ValueError("Must specify 'indptr', 'indices', and 'shape' arguments when passing data array.")
+                indptr = self.__check_arr(indptr, 'indptr')
+                indices = self.__check_arr(indices, 'indices')
+                shape = self.__check_arr(shape, 'shape')
                 if len(shape) != 2:
-                    raise ValueError('shape must specify two and only two dimensions')
+                    raise ValueError("'shape' argument must specify two and only two dimensions.")
                 data = sps.csr_matrix((data, indices, indptr), shape=shape)
             else:
-                raise ValueError("cannot use ndarray of dimensionality > 2")
+                raise ValueError("'data' argument cannot be ndarray of dimensionality > 2.")
         self.__data = data
-        self.__shape = data.shape
 
     @staticmethod
-    def __check_ind(ar, arg):
-        if not (ar.ndim == 1 or np.issubdtype(ar.dtype, int)):
-            raise ValueError('%s must be a 1D array of integers' % arg)
+    def __check_arr(ar, arg):
+        try:
+            ar = to_uint_array(ar)
+        except ValueError as ve:
+            raise ValueError("Cannot convert '%s' to an array of unsigned integers." % arg) from ve
+        if ar.ndim != 1:
+            raise ValueError("'%s' must be a 1D array of unsigned integers." % arg)
+        return ar
 
     def __getattr__(self, val):
-        return getattr(self.__data, val)
-
-    @property
-    def shape(self):
```

Comment:
Is there a particular reason for removing the ``shape`` property?

Reply:
It is redundant with `self.__getattr__('shape')` which defers to `self.__data.shape` (equal to before), but then the values are converted to uints.

Reply:
Thanks for the clarification. That makes sense.